server/views/applications/pages/required-referrals/accommodation-referral-details/dtr-details.njk

```diff
@@ -46,4 +46,34 @@
     ) }}
   {% endif %}
 
+  {% set itemsWithDetail = [] %}
+  {% set conditionalHtml %}
+    {{ formPageTextarea({
+      fieldName: "dutyToReferOutcomeOtherDetails",
+      label: {
+        text: "Add details about the reason",
+        classes: "govuk-label--s"
+      }
+    },fetchContext()) }}
+  {% endset %}
+
+  {% for item in page.items() %}
+    {% set itemsWithDetail = (
+      itemsWithDetail.push(mergeObjects(item, {
+        conditional: { html: conditionalHtml } if item.value === 'rejectedOther' else undefined
+      })), itemsWithDetail)
+    %}
+  {% endfor %}
```

Comment:
That's an interesting way of doing it -- I've updated the `convertObjectsToRadioItems` template plugin to take an optional argument to set conditional on the necessary items. Might be worth merging these once both are into main!